Makefile

```
@@ -291,13 +291,8 @@ clean-docker: ## Deletes the docker containers for local development.
 govet: ## Runs govet against all packages.
 	@echo Running GOVET
 	$(GO) get -u golang.org/x/tools/go/analysis/passes/shadow/cmd/shadow
-	$(GO) vet $(GOFLAGS) $(TE_PACKAGES) || exit 1
-	$(GO) vet -vettool=$(which shadow) $(GOFLAGS) $(TE_PACKAGES) || exit 1
-
-ifeq ($(BUILD_ENTERPRISE_READY),true)
-	$(GO) vet $(GOFLAGS) $(TE_PACKAGES) || exit 1
-	$(GO) vet -vettool=$(which shadow) $(GOFLAGS) $(EE_PACKAGES) || exit 1
-endif
+	$(GO) vet $(GOFLAGS) $(ALL_PACKAGES) || exit 1
+	$(GO) vet -vettool=$(GOPATH)/bin/shadow $(GOFLAGS) $(ALL_PACKAGES) || exit 1
```

You can simlity this to
```suggestion
	$(GO) vet -vettool=$(which shadow) $(GOFLAGS) $(TE_PACKAGES) || exit 1
```

---

Actually, no that's precisely what I'm fixing here: `which shadow` works on the command line, but `which` invoked from the `Makefile` context doesn't search `$PATH` and so doesn't find `shadow`. 

You can see this in `master` at the moment, in which `make govet` invokes:
```
go vet -vettool=  github.com/mattermost/mattermost-server/...
```

---

My bad, I did a copy&past error. Please take a second look. The magic is the second `$`.

---

Oh, right! That makes sense in terms of syntax. My secondary concern with `which` is that the user has modified `PATH` to include `$(GOPATH)/bin` -- I don't think this is "free" by default. Whereas using `$(GOPATH)/bin` directly is guaranteed to work -- at least until we switch to go modules.

Thoughts?

---

Well, https://golang.org/doc/install#tarball suggests to add `$(GOPATH)/bin` to `PATH`. I would say we can assume this.

My concern is, that I'm looking forward to go modules and want to prepare for this rather sooner then later. I'm 2/5 on not using `$GOPATH` and 1/5 on `which` is the right solution.

---

I agree that a developer having `$(GOPATH)/bin` in `PATH` is probable, but still has the "newbie" edge cases that `$GOPATH` won't -- and for whom `go vet` is probably most valuable. So /today/ `$(GOPATH)/bin` still seems like a net win.

I'm also keen on switching to go modules, but -- unless I haven't caught up on this point -- it's unclear to me at the moment where one-off tools on a module-by-module basis get "installed" as such. So my intuition is that `which` won't solve anything here. If it turns out that nothing has changed -- and there is still a global bin path -- then `$(GOPATH)/bin` will continue to work.

Ok to defer the go module details to that (holistic) switch?

---

You have good point. :+1:  Fine by me to defer this.